oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/json/JsopBuilder.java

```diff
@@ -275,6 +277,34 @@ public static String encode(String s) {
         return '\"' + s + '\"';
     }
 
+    /**
+     * Similar to #encode(String) but appends the result to the given buffer. This method exists to avoid the allocation
+     * of intermediate objects to represent the encoded String when the caller already has a buffer where the result
+     * should be appended.
+     *
+     * @param s the text to convert
+     * @param buff the target buffer
+     * @return the target buffer with the encoded string appended
+     */
+    private static StringBuilder encode(String s, StringBuilder buff) {
```

Comment:
maybe return the StringBuffer so this can be used fluently?

Reply:
The method does return the StringBuilder received as an argument.